src/org/thoughtcrime/securesms/qr/QrShowFragment.java

```diff
@@ -119,6 +129,31 @@ public void onReceive(Context context, Intent intent) {
         return view;
     }
 
+    public void shareQr() {
+        try {
+            File file = new File(getActivity().getExternalCacheDir(), "share-qr.png");
```

Comment:
any special reason to prefer [`getExternalCacheDir()`](https://developer.android.com/reference/android/content/Context.html#getExternalCacheDir%28%29) over [`getCacheDir`](https://developer.android.com/reference/android/content/Context#getCacheDir()) - they seem to be widely identical, however, only for the latter the system always takes care about deletion and cleaning up.

also, for the external cache, doc says "Shared storage may not always be available, since removable media can be ejected by the user".

Reply:
We use `getExternalCacheDir()` with preference over `getCacheDir()` in `PersistentBlobProvider.getExternalDir()` I will use that instead of using `getExternalCacheDir()` direclty

btw, I used `getExternalCacheDir()` because if I use `getCacheDir()` I get an error:

```
java.lang.IllegalArgumentException: Failed to find configured root that contains /data/data/chat.delta.lite/cache/shar
e-qr.png
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.core.content.FileProvider$SimplePathStrategy.getUriForFile(FileProvider.java:744)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.core.content.FileProvider.getUriForFile(FileProvider.java:418)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at org.thoughtcrime.securesms.util.FileProviderUtil.getUriFor(FileProviderUtil.java:19)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at org.thoughtcrime.securesms.qr.QrShowFragment.shareQr(QrShowFragment.java:141)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at org.thoughtcrime.securesms.qr.QrActivity.onOptionsItemSelected(QrActivity.java:145)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at android.app.Activity.onMenuItemSelected(Activity.java:4326)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.fragment.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:383)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.appcompat.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:228)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.appcompat.app.ToolbarActionBar$2.onMenuItemClick(ToolbarActionBar.java:65)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.appcompat.widget.Toolbar$1.onMenuItemClick(Toolbar.java:207)  
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.appcompat.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected(ActionMenuView.java:779)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:834)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:158)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:985)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:975)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.appcompat.widget.ActionMenuView.invokeItem(ActionMenuView.java:623)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at androidx.appcompat.view.menu.ActionMenuItemView.onClick(ActionMenuItemView.java:151)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at android.view.View.performClick(View.java:8160)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at android.widget.TextView.performClick(TextView.java:16222)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at android.view.View.performClickInternal(View.java:8137)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at android.view.View.access$3700(View.java:888)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at android.view.View$PerformClick.run(View.java:30236)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at android.os.Handler.handleCallback(Handler.java:938)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at android.os.Handler.dispatchMessage(Handler.java:99)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at android.os.Looper.loop(Looper.java:246)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at android.app.ActivityThread.main(ActivityThread.java:8587)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at java.lang.reflect.Method.invoke(Native Method)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
11-03 19:44:56.925  8562  8562 W QrShowFragment:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
```

Reply:
of switched to `getCacheDir()`, the error was because `<cache-path name="cache" path="." />` was missing in `file_provider_paths.xml`